### DISCUSSION TITLE
Media Mtx to Restream a series of cameras

### DISCUSSION THREAD

**Discussion by elliot1971** [bluenviron/mediamtx#2995] [Questions]
### Question

Hi All,
We are wanting to reduce load on our cameras by using MediaMTX.
We are having difficulty working out the correct way to set this up.

Each camera has a built in and functioning RTSP server and we can see the streams perfectly.  We have several client apps that need to display those streams but we often hit a stream limit on the camera itself.  

Can we use just MediaMTX or do we need to use FFMPeg for this restreaming so we make a single stream request to the camera and then connect to our restreamer for the client app usage?

Really appreciate any guidance.

**Comment by asturl** [comment]
Yes, MediaMTX can grab directly from the source and provide a restreamed endpoint "one to many"
For example, put a path in the config like this:
```
paths:
    restreamedcamera:
        source: rtsp://yourcameraip/rtspendpoint
```
And now you can open 
**rtsp://yourserver:8554/restreamedcamera** 
on any clients that can open rtsp. Or use the WebRtc functionality to show it in a browser client.

I've successfully used this with no added latency with a multitude of devices that provide RTSP streams, including one that can't handle more than a single stream.

**Comment by elliot1971** [comment]
Thanks,

My issue is the client apps are needing a H264 encoding which the camera gives when communicated with directly but when restreamed, the camera seems to pass MesiaMTX a h265 encoded stream which causes some issue, so I want to ensure I can restream from MediaMTX as h264 (until the client app is redeveloped to allow for h265.

Would I use ffmpeg and reencode on the fly? Any example source path that I can learn from?
Cheers

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by chinbhat** [reply]
You can run the ffmpeg command for runOnReady parameter.

Here is an example of how to configure the ffmpeg command and then you can adapt the command for your needs. MediaMTX will execute that command when the stream is ready. https://github.com/bluenviron/mediamtx/discussions/1896#discussioncomment-6242087

I assume that the server will use significant resources for transcoding, so provision your server accordingly.

**Reply by asturl** [reply]
Since reencoding is costly, I have to wonder about the call to your camera. This isn't really in the scope of MediaMTX, but getting a H265 stream with the **same endpoint** strikes me as strange - are there no parameters to control this?

**Reply by aler9** [reply]
from my experience, if you want to use WebRTC to view a IP camera, the simplest way consists into opening the camera settings and picking H264 instead of H265. Quality doesn't decrease, bandwidth is slightly higher. Leave re-encoding out of the game.

[a comment by an automated account is not shown]
